Unable to build on OSX 15 due to outdated libffi dependency in mpi

```console
error: failed to run custom build command for `libffi-sys v2.3.0`

Caused by:
  process didn't exit successfully: `/Users/~/development/Expander/target/debug/build/libffi-sys-b873ba362e74a81c/build-script-build` (exit status: 101)
  --- stdout
  checking build system type... aarch64-apple-darwin24.5.0
  checking host system type... aarch64-apple-darwin24.5.0
  checking target system type... aarch64-apple-darwin24.5.0
  continue configure in default builddir "./aarch64-apple-darwin"
  ....exec /bin/sh .././configure "--srcdir=.." "--enable-builddir=aarch64-apple-darwin" "darwin24.5.0"
  checking build system type... aarch64-apple-darwin24.5.0
  checking host system type... aarch64-apple-darwin24.5.0
  checking target system type... aarch64-apple-darwin24.5.0
  checking for gsed... sed
  checking for a BSD-compatible install... /opt/homebrew/bin/ginstall -c
  checking whether build environment is sane... yes
  checking for a race-free mkdir -p... /opt/homebrew/bin/gmkdir -p
  checking for gawk... no
  checking for mawk... no
  checking for nawk... no
  checking for awk... awk
  checking whether make sets $(MAKE)... yes
  checking whether make supports nested variables... yes
  checking for gcc... cc
  checking whether the C compiler works... yes
  checking for C compiler default output file name... a.out
  checking for suffix of executables... 
  checking whether we are cross compiling... no
  checking for suffix of object files... o
  checking whether the compiler supports GNU C... yes
  checking whether cc accepts -g... yes
  checking for cc option to enable C11 features... none needed
  checking whether cc understands -c and -o together... yes
  checking whether make supports the include directive... yes (GNU style)
  checking dependency style of cc... gcc3
  checking for g++... g++
  checking whether the compiler supports GNU C++... yes
  checking whether g++ accepts -g... yes
  checking for g++ option to enable C++11 features... none needed
  checking dependency style of g++... gcc3
  checking dependency style of cc... gcc3
  checking for grep that handles long lines and -e... /usr/bin/grep
  checking for egrep... /usr/bin/grep -E
  checking how to print strings... printf
  checking for a sed that does not truncate output... /usr/bin/sed
  checking for fgrep... /usr/bin/grep -F
  checking for ld used by cc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
  checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
  checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
  checking the name lister (/usr/bin/nm -B) interface... BSD nm
  checking whether ln -s works... yes
  checking the maximum length of command line arguments... 786432
  checking how to convert aarch64-apple-darwin24.5.0 file names to aarch64-apple-darwin24.5.0 format... func_convert_file_noop
  checking how to convert aarch64-apple-darwin24.5.0 file names to toolchain format... func_convert_file_noop
  checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
  checking for file... file
  checking for objdump... objdump
  checking how to recognize dependent libraries... pass_all
  checking for dlltool... no
  checking how to associate runtime and link libraries... printf %s\n
  checking for ar... ar
  checking for archiver @FILE support... no
  checking for strip... strip
  checking for ranlib... ranlib
  checking command to parse /usr/bin/nm -B output from cc object... ok
  checking for sysroot... no
  checking for a working dd... /bin/dd
  checking how to truncate binary pipes... /bin/dd bs=4096 count=1
  checking for mt... no
  checking if : is a manifest tool... no
  checking for dsymutil... dsymutil
  checking for nmedit... nmedit
  checking for lipo... lipo
  checking for otool... otool
  checking for otool64... no
  checking for -single_module linker flag... ld: warning: -single_module is obsolete
  no
  checking for -exported_symbols_list linker flag... yes
  checking for -force_load linker flag... yes
  checking for stdio.h... yes
  checking for stdlib.h... yes
  checking for string.h... yes
  checking for inttypes.h... yes
  checking for stdint.h... yes
  checking for strings.h... yes
  checking for sys/stat.h... yes
  checking for sys/types.h... yes
  checking for unistd.h... yes
  checking for dlfcn.h... yes
  checking for objdir... .libs
  checking if cc supports -fno-rtti -fno-exceptions... yes
  checking for cc option to produce PIC... -fno-common -DPIC
  checking if cc PIC flag -fno-common -DPIC works... yes
  checking if cc static flag -static works... no
  checking if cc supports -c -o file.o... yes
  checking if cc supports -c -o file.o... (cached) yes
  checking whether the cc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
  checking dynamic linker characteristics... darwin24.5.0 dyld
  checking how to hardcode library paths into programs... immediate
  checking whether stripping libraries is possible... yes
  checking if libtool supports shared libraries... yes
  checking whether to build shared libraries... no
  checking whether to build static libraries... yes
  checking how to run the C++ preprocessor... g++ -E
  checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
  checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
  checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
  checking for g++ option to produce PIC... -fno-common -DPIC
  checking if g++ PIC flag -fno-common -DPIC works... yes
  checking if g++ static flag -static works... no
  checking if g++ supports -c -o file.o... yes
  checking if g++ supports -c -o file.o... (cached) yes
  checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
  checking dynamic linker characteristics... darwin24.5.0 dyld
  checking how to hardcode library paths into programs... immediate
  checking for readelf... no
  checking size of size_t... 8
  checking for C compiler vendor... clang
  checking CFLAGS for most reasonable warnings... -Wall
  checking whether to enable maintainer-specific portions of Makefiles... no
  checking for sys/memfd.h... no
  checking for memfd_create... no
  checking for sys/mman.h... yes
  checking for mmap... yes
  checking for mkostemp... yes
  checking for mkstemp... yes
  checking for sys/mman.h... (cached) yes
  checking for mmap... (cached) yes
  checking whether read-only mmap of a plain file works... yes
  checking whether mmap from /dev/zero works... no
  checking for MAP_ANON(YMOUS)... yes
  checking whether mmap with MAP_ANON(YMOUS) works... yes
  checking for egrep... (cached) /usr/bin/grep -E
  checking for memcpy... yes
  checking for size_t... yes
  checking for working alloca.h... yes
  checking for alloca... yes
  checking size of double... 8
  checking size of long double... 8
  checking whether byte ordering is bigendian... no
  checking assembler .cfi pseudo-op support... yes
  checking whether compiler supports pointer authentication... no
  checking for _ prefix in compiled symbols... yes
  checking whether C compiler accepts -fno-lto... yes
  checking whether .eh_frame section should be read-only... yes
  checking for __attribute__((visibility("hidden")))... yes
  configure: versioning on shared library symbols is no
  checking that generated files are newer than configure... done
  configure: creating ./config.status
  config.status: creating include/Makefile
  config.status: creating include/ffi.h
  config.status: creating Makefile
  config.status: creating testsuite/Makefile
  config.status: creating man/Makefile
  config.status: creating doc/Makefile
  config.status: creating libffi.pc
  config.status: creating fficonfig.h
  config.status: executing buildir commands
  config.status: create top_srcdir/Makefile guessed from local Makefile
  config.status: build in aarch64-apple-darwin (HOST=aarch64-apple-darwin)
  config.status: executing depfiles commands
  config.status: executing libtool commands
  config.status: executing include commands
  config.status: executing src commands
  MAKE aarch64-apple-darwin :        1 * install
  Making install in include
  make[3]: Nothing to be done for `install-exec-am'.
   /opt/homebrew/bin/gmkdir -p '/Users/~/development/Expander/target/debug/build/libffi-sys-75016711cc5d026a/out/libffi-root/include'
   /opt/homebrew/bin/ginstall -c -m 644 ffi.h ffitarget.h '/Users/~/development/Expander/target/debug/build/libffi-sys-75016711cc5d026a/out/libffi-root/include'
  Making install in testsuite
  make[3]: Nothing to be done for `install-exec-am'.
  make[3]: Nothing to be done for `install-data-am'.
  Making install in man
  make[3]: Nothing to be done for `install-exec-am'.
   /opt/homebrew/bin/gmkdir -p '/Users/~/development/Expander/target/debug/build/libffi-sys-75016711cc5d026a/out/libffi-root/share/man/man3'
   /opt/homebrew/bin/ginstall -c -m 644 ../../man/ffi.3 ../../man/ffi_call.3 ../../man/ffi_prep_cif.3 ../../man/ffi_prep_cif_var.3 '/Users/~/development/Expander/target/debug/build/libffi-sys-75016711cc5d026a/out/libffi-root/share/man/man3'
  depbase=`echo src/prep_cif.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/prep_cif.lo -MD -MP -MF $depbase.Tpo -c -o src/prep_cif.lo ../src/prep_cif.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/prep_cif.lo -MD -MP -MF src/.deps/prep_cif.Tpo -c ../src/prep_cif.c  -fno-common -DPIC -o src/prep_cif.o
  depbase=`echo src/types.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/types.lo -MD -MP -MF $depbase.Tpo -c -o src/types.lo ../src/types.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/types.lo -MD -MP -MF src/.deps/types.Tpo -c ../src/types.c  -fno-common -DPIC -o src/types.o
  depbase=`echo src/raw_api.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/raw_api.lo -MD -MP -MF $depbase.Tpo -c -o src/raw_api.lo ../src/raw_api.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/raw_api.lo -MD -MP -MF src/.deps/raw_api.Tpo -c ../src/raw_api.c  -fno-common -DPIC -o src/raw_api.o
  depbase=`echo src/java_raw_api.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/java_raw_api.lo -MD -MP -MF $depbase.Tpo -c -o src/java_raw_api.lo ../src/java_raw_api.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/java_raw_api.lo -MD -MP -MF src/.deps/java_raw_api.Tpo -c ../src/java_raw_api.c  -fno-common -DPIC -o src/java_raw_api.o
  depbase=`echo src/closures.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/closures.lo -MD -MP -MF $depbase.Tpo -c -o src/closures.lo ../src/closures.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/closures.lo -MD -MP -MF src/.deps/closures.Tpo -c ../src/closures.c  -fno-common -DPIC -o src/closures.o
  depbase=`echo src/tramp.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/tramp.lo -MD -MP -MF $depbase.Tpo -c -o src/tramp.lo ../src/tramp.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/tramp.lo -MD -MP -MF src/.deps/tramp.Tpo -c ../src/tramp.c  -fno-common -DPIC -o src/tramp.o
  depbase=`echo src/aarch64/ffi.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src   -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/aarch64/ffi.lo -MD -MP -MF $depbase.Tpo -c -o src/aarch64/ffi.lo ../src/aarch64/ffi.c &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -Wall -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -fexceptions -MT src/aarch64/ffi.lo -MD -MP -MF src/aarch64/.deps/ffi.Tpo -c ../src/aarch64/ffi.c  -fno-common -DPIC -o src/aarch64/ffi.o
  depbase=`echo src/aarch64/sysv.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src  -I. -I../include -Iinclude -I../src -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -MT src/aarch64/sysv.lo -MD -MP -MF $depbase.Tpo -c -o src/aarch64/sysv.lo ../src/aarch64/sysv.S &&\
        mv -f $depbase.Tpo $depbase.Plo
  libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -I. -I../include -Iinclude -I../src -O0 -ffunction-sections -fdata-sections -fPIC -gdwarf-2 -fno-omit-frame-pointer --target=arm64-apple-macosx -mmacosx-version-min=15.5 -Wno-implicit-function-declaration -MT src/aarch64/sysv.lo -MD -MP -MF src/aarch64/.deps/sysv.Tpo -c ../src/aarch64/sysv.S  -fno-common -DPIC -o src/aarch64/sysv.o

  --- stderr
  rm: conftest.dSYM: is a directory
  rm: conftest.dSYM: is a directory
  rm: conftest.dSYM: is a directory
  rm: conftest.dSYM: is a directory
  clang: error: unsupported option '-print-multi-os-directory'
  clang: error: no input files
  ../src/java_raw_api.c:328:46: warning: 'ffi_java_raw_size' is deprecated [-Wdeprecated-declarations]
    328 |   ffi_java_raw *raw = (ffi_java_raw*)alloca (ffi_java_raw_size (cif));
        |                                              ^
  include/ffi.h:326:56: note: 'ffi_java_raw_size' has been explicitly marked deprecated here
    326 | size_t ffi_java_raw_size (ffi_cif *cif) __attribute__((deprecated));
        |                                                        ^
  ../src/java_raw_api.c:331:3: warning: 'ffi_java_ptrarray_to_raw' is deprecated [-Wdeprecated-declarations]
    331 |   ffi_java_ptrarray_to_raw (cif, avalue, raw);
        |   ^
  include/ffi.h:322:93: note: 'ffi_java_ptrarray_to_raw' has been explicitly marked deprecated here
    322 | void ffi_java_ptrarray_to_raw (ffi_cif *cif, void **args, ffi_java_raw *raw) __attribute__((deprecated));
        |                                                                                             ^
  2 warnings generated.
  /var/folders/f1/9ynx2f255g504jrj2tq0z3540000gn/T/sysv-782c7a.s:27:2: error: invalid CFI advance_loc expression
   .cfi_def_cfa x1, 40;
   ^
  /var/folders/f1/9ynx2f255g504jrj2tq0z3540000gn/T/sysv-782c7a.s:188:2: error: invalid CFI advance_loc expression
   .cfi_adjust_cfa_offset (8*2 + (8 * 16 + 8 * 8) + 64)
   ^
  make[2]: *** [src/aarch64/sysv.lo] Error 1
  make[1]: *** [install-recursive] Error 1
  make: *** [install] Error 2

  thread 'main' panicked at /Users/~/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libffi-sys-2.3.0/build/common.rs:8:5:
  Building libffi
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
The issue was already fixed in mpi: https://github.com/rsmpi/rsmpi/pull/214

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ quote = "1.0"     # For generating code
 proc-macro2 = "1.0"  # For working with tokens
 itertools = "0.13"
 log = "0.4"
-mpi = "0.8.0"
+mpi = { git = "https://github.com/rsmpi/rsmpi", rev = "61796831954b679cbe267c1b704ddbcb7fef3715" }
 rand = "0.8.5"
 raw-cpuid = "11.1.0"
 rayon = "1.10"

--- a/gkr_engine/src/mpi_engine/definition.rs
+++ b/gkr_engine/src/mpi_engine/definition.rs
@@ -1,8 +1,5 @@
 use arith::Field;
-use mpi::{
-    ffi::{ompi_win_t, MPI_Win_free},
-    topology::Process,
-};
+use mpi::{ffi::ompi_win_t, topology::Process};
 use serdes::ExpSerde;
 
 use super::MPISharedMemory;
@@ -166,7 +163,9 @@ pub trait MPIEngine {
     /// Discard the control of shared memory segment
     fn free_shared_mem(&self, window: &mut *mut ompi_win_t) {
         unsafe {
-            MPI_Win_free(window as *mut *mut ompi_win_t);
+            // Reconstruct an MPI_Win handle from the raw pointer and let MPI release it.
+            let mut win_handle = mpi::ffi::MPI_Win(*window);
+            mpi::ffi::MPI_Win_free(&mut win_handle as *mut mpi::ffi::MPI_Win);
         }
     }
 }

--- a/gkr_engine/src/mpi_engine/engine.rs
+++ b/gkr_engine/src/mpi_engine/engine.rs
@@ -429,7 +429,10 @@ impl<'a> MPIEngine for MPIConfig<'a> {
     fn create_shared_mem(&self, n_bytes: usize) -> (*mut u8, *mut ompi_win_t) {
         let window_size = if self.is_root() { n_bytes } else { 0 };
         let mut baseptr: *mut c_void = std::ptr::null_mut();
-        let mut window = std::ptr::null_mut();
+
+        // Handle to the MPI window.  Initialize to null and let MPI fill it in.
+        let mut window_handle: MPI_Win = MPI_Win(std::ptr::null_mut());
+
         unsafe {
             MPI_Win_allocate_shared(
                 window_size as isize,
@@ -437,26 +440,26 @@ impl<'a> MPIEngine for MPIConfig<'a> {
                 RSMPI_INFO_NULL,
                 self.world.unwrap().as_raw(),
                 &mut baseptr as *mut *mut c_void as *mut c_void,
-                &mut window,
+                &mut window_handle as *mut MPI_Win,
             );
             self.barrier();
 
             if !self.is_root() {
-                let mut size = 0;
-                let mut disp_unit = 0;
+                let mut size: MPI_Aint = 0;
+                let mut disp_unit: ::std::os::raw::c_int = 0;
                 let mut query_baseptr: *mut c_void = std::ptr::null_mut();
                 MPI_Win_shared_query(
-                    window,
+                    window_handle,
                     0,
-                    &mut size,
-                    &mut disp_unit,
+                    &mut size as *mut MPI_Aint,
+                    &mut disp_unit as *mut ::std::os::raw::c_int,
                     &mut query_baseptr as *mut *mut c_void as *mut c_void,
                 );
                 baseptr = query_baseptr;
             }
         }
 
-        (baseptr as *mut u8, window)
+        (baseptr as *mut u8, window_handle.0)
     }
 }
 

